### ISSUE TITLE
ERRATA - Flavors of attention in modern diffusion models

### ISSUE THREAD

**Issue by tolgacangoz** [sayakpaul/portfolio#22]
Thanks for the great blog post @sayakpaul !

1. I guess there is a missing citation for Lumina-Next [[zhuo2024luminanextmakingluminat2xstronger?](https://sayak.dev/posts/attn-diffusion.html#ref-zhuo2024luminanextmakingluminat2xstronger)]
2. `    self.self.qkv_mlp_first(x_mod), `

**Comment by tolgacangoz** [comment]
I might be wrong: concatenation part of the joint attention is:
```py
        # Concatenate
        q = torch.cat([context_q, q])
        k = torch.cat([context_k, k])
        v = torch.cat([context_v, v])
```
Aren't concatenations supposed to be done in the sequence length dimension rather than the batch dimension? I thought maybe you didn't consider this, but other codes have a `dim` parameter set.

**Comment by sayakpaul** [comment]
Thanks so much. Pushed some fixes.

**Comment by tolgacangoz** [comment]
Thanks for considering!

**Comment by tolgacangoz** [comment]
Additionally, I guess at the linear attention:
```
+    # Combine scores
+    scores = torch.matmul(v.transpose(-1, -2), k)
+    x = torch.matmul(scores, q.transpose(-1, -2))
```

**Comment by sayakpaul** [comment]
Thanks once again! Fixed.
